### PR TITLE
Update optimize-selectors.md

### PR DESCRIPTION
**Pull request by adardesign**

### CODE REVIEW

File: page/performance/optimize-selectors.md [hunk 1]
```diff
@@ -58,3 +58,13 @@ $( ".category :radio" ); // Implied universal selection.
...
 $( ".category input:radio" ); // Much better.
 ```
+
+## Avoid using direct attribute Selectors
+
+Attribute selectors could be very slow.
+
+```
+$("[attributeKey=value]"); // Extremely expensive.
+$("input[attributeKey=value]");; // Better.
```
Comment (arthurvr) [inline page/performance/optimize-selectors.md]: two semicolons

File: page/performance/optimize-selectors.md [hunk 2]
```diff
@@ -58,3 +58,13 @@ $( ".category :radio" ); // Implied universal selection.
 $( ".category *:radio" ); // Same thing, explicit now.
 $( ".category input:radio" ); // Much better.
 ```
+
+## Avoid using direct attribute Selectors
+
+Attribute selectors could be very slow.
+
+```
+$("[attributeKey=value]"); // Extremely expensive.
```
Comment (arthurvr) [inline page/performance/optimize-selectors.md]: Attribute key? Attributes have [a name](http://www.w3.org/TR/html-markup/syntax.html#attribute-name).

File: page/performance/optimize-selectors.md [hunk 3]
```diff
@@ -58,3 +58,13 @@ $( ".category :radio" ); // Implied universal selection.
 $( ".category *:radio" ); // Same thing, explicit now.
 $( ".category input:radio" ); // Much better.
 ```
+
+## Avoid using direct attribute Selectors
+
+Attribute selectors could be very slow.
```
Comment (arthurvr) [inline page/performance/optimize-selectors.md]: People are also interested in reading why and how to improve it.

File: page/performance/optimize-selectors.md [hunk 4]
```diff
@@ -58,3 +58,13 @@ $( ".category :radio" ); // Implied universal selection.
 $( ".category *:radio" ); // Same thing, explicit now.
 $( ".category input:radio" ); // Much better.
 ```
+
+## Avoid using direct attribute Selectors
```
Comment (arthurvr) [inline page/performance/optimize-selectors.md]: Could you fix the capitalization of this heading?

File: page/performance/optimize-selectors.md [hunk 5]
```diff
@@ -58,3 +58,13 @@ $( ".category :radio" ); // Implied universal selection.
...
+
+Attribute selectors could be very slow.
+
+```
+$("[attributeKey=value]"); // Extremely expensive.
+$("input[attributeKey=value]");; // Better.
+$(".formEle").find("input[attributeKey=value]"); //  Much better.
```
Comment (arthurvr) [inline page/performance/optimize-selectors.md]: > formEle

`Ele` seems like a weird abbreviation.

File: page/performance/optimize-selectors.md [hunk 6]
```diff
@@ -58,3 +58,13 @@ $( ".category :radio" ); // Implied universal selection.
 $( ".category *:radio" ); // Same thing, explicit now.
 $( ".category input:radio" ); // Much better.
 ```
+
+## Avoid using direct attribute Selectors
+
+Attribute selectors could be very slow.
+
+```
```
Comment (arthurvr) [inline page/performance/optimize-selectors.md]: Could you turn on markdown syntax highlighting?

**Comment by arthurvr** [comment]
Could you please sign the [CLA](http://contribute.jquery.org/CLA/)?

**Comment by arthurvr** [comment]
Could you take a look at https://github.com/jquery/learn.jquery.com/issues/106? I'd love to include a part about attribute selectors, but there are some more things we need to touch. That said, could you update the examples to use our [style guide?](http://contribute.jquery.org/style-guide/js/)

File: page/performance/optimize-selectors.md (hunk 5 above)
Comment (adardesign) [inline page/performance/optimize-selectors.md]: What can be a better name for a arbitrary element?

Comment (kswedberg) [inline page/performance/optimize-selectors.md]: How about just `$("form")`? Inputs are typically going to be within the context of a form, right?

Comment (adardesign) [inline page/performance/optimize-selectors.md]: good point, thanks

**Comment by arthurvr** [comment]
Hey @adardesign, are you interested in finishing up this PR? You still need to sign the CLA and address some comments.

**Comment by adardesign** [comment]
Yes, Ill get to it shortly

**Comment by arthurvr** [comment]
Hey-hey @adardesign still interested?

File: page/performance/optimize-selectors.md (hunk 2 above)
Comment (adardesign) [inline page/performance/optimize-selectors.md]: @arthurvr I did not understand what you meant here, Please explain.

File: page/performance/optimize-selectors.md (hunk 4 above)
Comment (adardesign) [inline page/performance/optimize-selectors.md]: Fixed!

File: page/performance/optimize-selectors.md (hunk 3 above)
Comment (adardesign) [inline page/performance/optimize-selectors.md]: Added explanation

File: page/performance/optimize-selectors.md (hunk 1 above)
Comment (adardesign) [inline page/performance/optimize-selectors.md]: Fixed!

File: page/performance/optimize-selectors.md (hunk 5 above)
Comment (adardesign) [inline page/performance/optimize-selectors.md]: Done!

**Comment by arthurvr** [comment]
Friendly ping @adardesign. Don't forget about the CLA too.

**Comment by arthurvr** [comment]
bump @adardesign

**Comment by AurelioDeRosa** [comment]
Hi @adardesign. Are you still interested in this PR?

**Comment by AurelioDeRosa** [comment]
Thinking at this PR again, I think it isn't relevant anymore. From a long time jQuery uses `querySelector` in supporting browsers which is able to deal with the attribute selector. So, it isn't expansive as it used to be. I reckon this PR can be closed. Any thoughts?

**Comment by dmethvin** [comment]
Agreed, `querySelectorAll` is generally fast enough and if there are performance issues it would be better to diagnose a slow page than to attempt premature optimization.

**Pull request closed without merging**